src/frontend/src/instance/distributed.rs

```diff
@@ -569,36 +584,21 @@ impl DistInstance {
             .context(RequestMetaSnafu)
     }
 
-    // TODO(LFC): Refactor GRPC insertion and deletion implementation here,
-    // Take insertion as an example. GRPC insertion is converted to Table InsertRequest here,
-    // than split the Table InsertRequest in DistTable, than assemble each GRPC InsertRequest there.
-    // Rather inefficient, should operate on GRPC InsertRequest directly.
     async fn handle_dist_insert(
         &self,
-        request: InsertRequest,
+        requests: InsertRequests,
         ctx: QueryContextRef,
     ) -> Result<Output> {
-        let catalog = &ctx.current_catalog();
-        let schema = &ctx.current_schema();
-        let table_name = &request.table_name;
-        let table_ref = TableReference::full(catalog, schema, table_name);
-
-        let table = self
-            .catalog_manager
-            .table(catalog, schema, table_name)
-            .await
-            .context(CatalogSnafu)?
-            .with_context(|| TableNotFoundSnafu {
-                table_name: table_ref.to_string(),
-            })?;
-
-        let request = common_grpc_expr::insert::to_table_insert_request(catalog, schema, request)
-            .context(ToTableInsertRequestSnafu)?;
-
-        let affected_rows = table.insert(request).await.context(TableSnafu)?;
-        Ok(Output::AffectedRows(affected_rows))
+        let inserter = DistInserter::new(
+            ctx.current_catalog(),
+            ctx.current_schema(),
+            self.catalog_manager.clone(),
+        );
+        let affected_rows = inserter.grpc_insert(requests).await?;
+        Ok(Output::AffectedRows(affected_rows as usize))
     }
 
+    // TODO(LFC): Like insertions above, refactor GRPC deletion impl here.
```

Comment:
There are a lot of `TODO` in this PR, we can create some issues to track them.

Reply:
will find some time to do that for all my todos